### PR TITLE
Update README.md to include branch matching information

### PR DESCRIPTION
**Pull request by lmbr-pip**
Addresses https://github.com/o3de/o3de-multiplayersample/issues/184

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +57,8 @@ These instructions use the following installation paths. Be sure to substitute y
 
 <span style="background-color:#4F3C3C">**NOTE:** You can clone the project to any local directory. If you clone the project inside an existing Git repository directory (for example, the directory that contains your local O3DE engine repository) you should add the o3de-multiplayersample project directory to the Git exclude file for the existing Git repository.</span>
 
+It is expected that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same branch. If using development o3de-multiplayersample, then match with development branches of o3de, and o3de-multiplayersample. If using a release version of o3de, then checkout the corresponding tags for o3de-multiplayer and o3de-multiplayersample-assets ie `git checkout tags/<tag> -b local branch name>`.
```
Comment (allisaurus) [inline README.md:60]: this is very useful info, but I feel like it's too easy to miss up here. Most folks will just jump to the steps (speaking from personal experience, anyway)

what do you think about leaving this but also adding a 5th step to both clone options like: 
"Ensure your o3de, o3de-multiplayersample, and o3de-multiplayersample-assets repos are all checked out on the same branch (such as: `development`). See above note for details."

Comment (mbalfour-amzn) [reply README.md:60]: +1 
I'd take it even a little bit further, and just have a step after cloning where you do the git checkout and then reupdate the submodule, and put the description of the choice of branches (development vs stabilization) there.

File: README.md [hunk 2]
```diff
@@ -117,6 +117,12 @@ These instructions use the following installation paths. Be sure to substitute y
    echo o3de-multiplayersample-assets > C:/o3de/.git/info/exclude
    ```
 
+## Step 1a. Ensure your branches match
+
+Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets, when switching branches in that repository.
+
+If you're using an installed version of o3de, then checkout the corresponding versions of the sample reprositories that match the release. O3DE uses tags to identify the release compatible version of each repository. There will be match release tags for o3de-multiplayer and o3de-multiplayersample-assets which can be found via https://github.com/o3de/o3de-multiplayersample/tags. Branches can be checked out using Git's tag syntax, `git checkout tags/<tag> -b local branch name>`.
```
Comment (lmbr-pip) [inline README.md:124]:
```suggestion
If you're using a release or installer version of O3DE, then checkout the corresponding versions of the sample repositories that match the release. O3DE uses [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) to identify the release compatible version of each repository. There will be release tags set for o3de-multiplayer and o3de-multiplayersample-assets, which can be found via https://github.com/o3de/o3de-multiplayersample/tags. Branches can be checked out using standard Git commands for example, `git checkout tags/<tag> -b local branch name`.
```

File: README.md [hunk 3]
```diff
@@ -117,6 +117,16 @@ These instructions use the following installation paths. Be sure to substitute y
    echo o3de-multiplayersample-assets > C:/o3de/.git/info/exclude
    ```
 
+## Step 1a. Ensure your branches match
+
+Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets, when switching branches in that repository.
```
Comment (mbalfour-amzn) [inline README.md:122]:
```suggestion
Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets when switching branches in that repository.
```

File: README.md [hunk 4]
```diff
@@ -117,6 +117,16 @@ These instructions use the following installation paths. Be sure to substitute y
...
+
+If you're using a release or installer version of O3DE, then you must checkout versions of the sample repositories that match the release. O3DE uses standard Git [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) to identify the release compatible version of each repository.
```
Comment (mbalfour-amzn) [inline README.md:124]:
```suggestion
If you're using a release or installer version of O3DE, then you must checkout versions of the sample repositories that match the release. O3DE uses standard Git [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) to identify the release-compatible version of each repository. 
```

File: README.md [hunk 5]
```diff
@@ -117,6 +117,16 @@ These instructions use the following installation paths. Be sure to substitute y
    echo o3de-multiplayersample-assets > C:/o3de/.git/info/exclude
    ```
 
+## Step 1a. Ensure your branches match
+
+Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets, when switching branches in that repository.
+
+If you're using a release or installer version of O3DE, then you must checkout versions of the sample repositories that match the release. O3DE uses standard Git [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) to identify the release compatible version of each repository. 
+
+For each O3DE release, repositories that have been updated to match the release, should have a matching tag for the release. You can see all defined tags using the [Tags](https://github.com/o3de/o3de-multiplayersample/tags] view in each repository.
```
Comment (mbalfour-amzn) [inline README.md:126]:
```suggestion
For each O3DE release, repositories that have been updated to match the release should have a matching tag for the release. You can see all defined tags using the [Tags](https://github.com/o3de/o3de-multiplayersample/tags] view in each repository.
```

**Review by mbalfour-amzn — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -117,6 +117,16 @@ These instructions use the following installation paths. Be sure to substitute y
    echo o3de-multiplayersample-assets > C:/o3de/.git/info/exclude
    ```
 
+## Step 1a. Ensure your branches match
+
+Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets when switching branches in that repository.
```
Comment (allisaurus) [inline README.md:122]:
```suggestion
Before building the project, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample-assets. Ensure that you update the submodules in o3de-multiplayersample-assets when switching branches in that repository.
```

**Review by allisaurus — APPROVED** [review]

**Pull request merged**